### ISSUE TITLE
Using deprecated access of private `pytest` class

### ISSUE THREAD

**Issue by tyler-jachetta** [taverntesting/tavern#655]
https://github.com/taverntesting/tavern/blob/69e8cf4db7c067f94d3ad480bd8295c1c6c335a6/tavern/testutils/pytesthook/item.py#L57

I'm not sure as of when, but at least using `pytest` 6.2.2, `tavern` is causing this warning. I _think_ that it's due to directly invoking `_pytest.fixtures.FixtureRequest` directly, though I'm not 100% certain.

```
 =============================== warnings summary ===============================
../../../.cache/pypoetry/virtualenvs/nexus-vcl-2IZOCotG-py3.8/lib/python3.8/site-packages/_pytest/config/__init__.py:1233
  /home/runner/.cache/pypoetry/virtualenvs/nexus-vcl-2IZOCotG-py3.8/lib/python3.8/site-packages/_pytest/config/__init__.py:1233: PytestConfigWarning: Unknown config option: tavern-beta-new-traceback
  
    self._warn_or_fail_if_strict(f"Unknown config option: {key}\n")

../../../.cache/pypoetry/virtualenvs/nexus-vcl-2IZOCotG-py3.8/lib/python3.8/site-packages/tavern/testutils/pytesthook/item.py:57
  /home/runner/.cache/pypoetry/virtualenvs/nexus-vcl-2IZOCotG-py3.8/lib/python3.8/site-packages/tavern/testutils/pytesthook/item.py:57: PytestDeprecationWarning: A private pytest class or function was used.
    self._request = fixtures.FixtureRequest(self)
```

**Comment by michaelboulton** [comment]
Fixed in 1.14.2-

**Comment by tyler-jachetta** [comment]
Thanks! :D
